File: typescript/template.ts
import fs from "fs";
import assert from "assert";
import _ from "lodash";
import { GridMap, GridSet } from "../aoc/utilities";

// #####
// INPUT
// #####
const parseInput = (puzzleInput: string) => {
  return puzzleInput.split("\n").map(Number);
};

const TEST_INPUT = ``;

// const puzzleInput = fs
//   .readFileSync("2021/data/day01_input.txt")
//   .toString()
//   .trim();

// ########
// SOLUTION
// ########

// part 1
const part1 = (puzzleInput: string) => {
  const tbd = parseInput(puzzleInput);
};
console.log(part1(TEST_INPUT));
// assert(part1(TEST_INPUT) === );
console.time("part 1");
// console.log(part1(puzzleInput));
// console.timeEnd("part 1");

// part 2
// const part2 = (puzzleInput: string) => {
//   const tbd = parseInput(puzzleInput);
// };
// console.log(part2(TEST_INPUT));
// assert(part2(TEST_INPUT) === );
// console.time("part 2");
// console.log(part2(puzzleInput));
// console.timeEnd("part 2");

File: 2016/template.dart
import 'dart:io';
import 'dart:math';

// ########
// SOLUTION
// ########
List<String> parseInput(String puzzleInput) {
  return puzzleInput.split("\n").map((item) => item.trim()).toList();
}

int part1(puzzleInput) {
  var data = parseInput(puzzleInput);
  return 0;
}

int part2(puzzleInput) {
  var data = parseInput(puzzleInput);
  return 0;
}

// ###########
// RUN PROGRAM
// ###########
void main() {
  String puzzleInput = File('data/dayXX_input.txt').readAsStringSync();
  const TEST_INPUT = "";

  // part 1
  assert(part1(TEST_INPUT) == 5);

  // final stopwatchPart1 = Stopwatch()..start();
  // print("part 1: ${part1(puzzleInput)}");
  // stopwatchPart1.stop();
  // print("Elapsed time: ${stopwatchPart1.elapsed}");

  // part 2
  // assert(part2(TEST_INPUT) == 4);

  // final stopwatchPart2 = Stopwatch()..start();
  // print("part 2: ${part2(puzzleInput)}");
  // stopwatchPart2.stop();
  // print("Elapsed time: ${stopwatchPart2.elapsed}");
}
